packages/framework-types/src/effect/index.ts

```diff
@@ -1,19 +1,849 @@
+ *
+ * type Either<TError, TResult> =
+ *   | { _tag: 'Left', value: TError }
+ *   | { _tag: 'Right', value: TResult}
+ *
+ * This means that an object of type Either could either be an object with the _tag field set to 'Left', and the value to be of type 'TError',
+ * or an object with the _tag field set to 'Right', and the value to be of type 'TResult'.
+ *
+ * In some programming languages it is called Result. Effect-ts uses the name Either, because it is more generic and it is not tied to
+ * represent a result of a computation.
+ *
+ * This is useful to represent when a function could return an error that we want to recover from, or a result that we want to use. Instead of
+ * throwing an error, we can return an Either with the error, and then recover from it. This is useful because the recovery logic can be completely
+ * handled by ourselves, and design the error recovery flow as we want, instead of catching and rethrowing errors.
+ *
+ * Takeaway: `Either` allows representing computations that can fail explicitly, ensuring that the caller of the function handles the error case.
+ */
+
+/**
+ * Type: Lazy
+ * ==========
+ *
+ * This type is not reexported because it is not necessary for the usage of effect, but it is useful that you
+ * know that it exists and how it works. It is defined like:
+ *
+ * type Lazy<T> = () => T
+ *
+ * This means that an object of type Lazy is a function that returns a value of type T. It is useful to represent
+ * a value that is not computed until it is needed, and it is useful to represent a value that is computed lazily.
+ *
+ * For example, if we have a function that returns a value of type T, but it is expensive to compute, we can
+ * represent it as a Lazy<T>, and then only compute it when it is needed.
+ *
+ * Effect-ts uses Lazy under the hoods to ensure performance and allow for better concurrency management.
+ *
+ * Takeaway: `Lazy` allows representing a value that is not computed until it is needed.
+ */
+
+/**
+ * Type: Effect
+ * ============
+ *
+ * The `Effect` type is the foundation of the `effect-ts` library. It is a type that represents a computation that
+ * has some kind of dependency, can fail, and that can be interrupted. It is defined like:
+ *
+ * type Effect<TEnv, TErr, TRes> = (dependencies: TEnv) => Lazy<Promise<Either<TErr, TRes>>>
+ *
+ * It essentially is a function that accepts an object of dependencies, and returns a function that returns a Promise
+ * that can either return an error or a result. It is generic so you can define the dependencies, the error, and the result.
+ *
+ * Effect-ts calls dependencies "environment", and it is a way to represent the dependencies of a computation. It is useful
+ * because it allows us to define the dependencies of a program in a single place, and then pass them at the top level of the
+ * application. It is also useful for unit testing, because we can pass a mocked environment to the computation, and then
```

Comment:
```suggestion
 * because it allows us to define the dependencies of a program in a single place, and then pass them to the top level of the
 * application. It is also useful for unit testing since we can pass a mocked environment to the computation, and then
```